controllers/hasapplication_controller_test.go

```diff
@@ -0,0 +1,265 @@
+			hasAppLookupKey := types.NamespacedName{Name: HASAppName, Namespace: HASAppNamespace}
+			createdHasApp := &appstudiov1alpha1.HASApplication{}
+
+			Eventually(func() bool {
+				k8sClient.Get(context.Background(), hasAppLookupKey, createdHasApp)
+				return len(createdHasApp.Status.Conditions) > 0
+			}, timeout, interval).Should(BeTrue())
+
+			// Make sure the devfile model was properly set
+			Expect(createdHasApp.Status.Devfile).Should(Not(Equal("")))
+
+			devfile, err := devfile.ParseDevfileModel(createdHasApp.Status.Devfile)
+			Expect(err).Should(Not(HaveOccurred()))
+
+			// gitOpsRepo and appModelRepo should both be set
+			Expect(string(devfile.GetMetadata().Attributes["appModelRepository.url"].Raw)).Should((ContainSubstring(hasApp.Spec.AppModelRepository.URL)))
+			Expect(string(devfile.GetMetadata().Attributes["gitOpsRepository.url"].Raw)).Should(Not(Equal("")))
+			Expect(string(devfile.GetMetadata().Attributes["gitOpsRepository.url"].Raw)).Should(Not(ContainSubstring(hasApp.Spec.AppModelRepository.URL)))
+
+			// Delete the specified resource
+			deleteHASAppCR(hasAppLookupKey)
+		})
+	})
+
+	Context("Update HASApplication CR fields", func() {
+		It("Should update successfully with updated description", func() {
+			ctx := context.Background()
+
+			hasApp := &appstudiov1alpha1.HASApplication{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "appstudio.redhat.com/v1alpha1",
+					Kind:       "HASApplication",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      HASAppName,
+					Namespace: HASAppNamespace,
+				},
+				Spec: appstudiov1alpha1.HASApplicationSpec{
+					DisplayName: DisplayName,
+					Description: Description,
+					AppModelRepository: appstudiov1alpha1.HASApplicationGitRepository{
+						URL: "https://github.com/testorg/petclinic-app",
+					},
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, hasApp)).Should(Succeed())
+
+			// Look up the has app resource that was created.
+			// num(conditions) may still be < 1 on the first try, so retry until at least _some_ condition is set
+			hasAppLookupKey := types.NamespacedName{Name: HASAppName, Namespace: HASAppNamespace}
+			fetchedHasApp := &appstudiov1alpha1.HASApplication{}
+			Eventually(func() bool {
+				k8sClient.Get(context.Background(), hasAppLookupKey, fetchedHasApp)
+				return len(fetchedHasApp.Status.Conditions) > 0
+			}, timeout, interval).Should(BeTrue())
+
+			// Update the name and description of the hasApp resource
+			fetchedHasApp.Spec.DisplayName = "newname"
+			fetchedHasApp.Spec.Description = "New Description"
+			Expect(k8sClient.Update(context.Background(), fetchedHasApp)).Should(Succeed())
+
+			// Make sure the devfile model was properly set
+			Expect(fetchedHasApp.Status.Devfile).Should(Not(Equal("")))
+
+			// Get the updated resource
+			Eventually(func() bool {
+				k8sClient.Get(context.Background(), hasAppLookupKey, fetchedHasApp)
+				return len(fetchedHasApp.Status.Conditions) > 1
```

Comment:
I wonder if its worth checking the `Updated` status here

Reply:
Good idea. I've updated it so that it'll return only if the most recent status condition is set to `Updated`.